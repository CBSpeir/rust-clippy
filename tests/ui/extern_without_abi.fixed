//@aux-build:proc_macros.rs

#![warn(clippy::extern_without_abi)]

extern crate proc_macros;
use proc_macros::{external, with_span};

#[rustfmt::skip]
extern "C" fn foo() {}
//~^ ERROR: `extern` missing explicit ABI

#[rustfmt::skip]
extern "C" 	
fn foo_two() {}
//~^^ ERROR: `extern` missing explicit ABI

extern "C" fn bar() {}

#[rustfmt::skip]
extern 	
"C" 
fn bar_two() {}

extern "system" fn baz() {}

#[rustfmt::skip]
extern "C" {
//~^ ERROR: `extern` missing explicit ABI
    fn qux();
}

#[rustfmt::skip]
extern "C" 	
{
//~^^ ERROR: `extern` missing explicit ABI
    fn qux_two();
}

#[rustfmt::skip]
extern "C" {fn qux_three();}
//~^ ERROR: `extern` missing explicit ABI

extern "C" {
    fn grault();
}

extern "system" {
    fn grault_two();
}

external! {
    extern fn waldo() {}
}

with_span! {
    span
    extern fn waldo_two() {}
}

fn main() {}
