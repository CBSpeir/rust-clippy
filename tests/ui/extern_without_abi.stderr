error: `extern` missing explicit ABI
  --> tests/ui/extern_without_abi.rs:9:1
   |
LL | extern fn foo() {}
   | ^^^^^^^^^^^^^^^^^^ help: consider using: `extern "C" fn foo() {}`
   |
   = note: `-D clippy::extern-without-abi` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::extern_without_abi)]`

error: `extern` missing explicit ABI
  --> tests/ui/extern_without_abi.rs:13:1
   |
LL | / extern     
LL | | fn foo_two() {}
   | |_______________^
   |
help: consider using
   |
LL ~ extern "C"     
LL + fn foo_two() {}
   |

error: `extern` missing explicit ABI
  --> tests/ui/extern_without_abi.rs:27:1
   |
LL | / extern {
LL | |
LL | |     fn qux();
LL | | }
   | |_^
   |
help: consider using
   |
LL + extern "C" {
LL +
LL +     fn qux();
LL + }
   |

error: `extern` missing explicit ABI
  --> tests/ui/extern_without_abi.rs:33:1
   |
LL | / extern     
LL | | {
LL | |
LL | |     fn qux_two();
LL | | }
   | |_^
   |
help: consider using
   |
LL ~ extern "C"     
LL + {
LL +
LL +     fn qux_two();
LL + }
   |

error: `extern` missing explicit ABI
  --> tests/ui/extern_without_abi.rs:40:1
   |
LL | extern {fn qux_three();}
   | ^^^^^^^^^^^^^^^^^^^^^^^^ help: consider using: `extern "C" {fn qux_three();}`

error: aborting due to 5 previous errors

